test/run.sh

```diff
@@ -1,77 +1,48 @@
-${ts} 2 1270.0.01/1234 forwarded ChEcKiP.DyNdNs.OrG to 2620:0:ccd::2
-${ts} 2 1270.0.01/1234 forwarded ChEcKiP.DyNdNs.OrG to 2620:0:ccc::2
-${ts} 2 1270.0.01/1234 reply ChEcKiP.DyNdNs.OrG is <CNAME>
-${ts} 2 1270.0.01/1234 reply ChEcKiP.DyNdNs.OrG is 216.146.38.70
-${ts} 2 1270.0.01/1234 reply ChEcKiP.DyNdNs.OrG is 216.146.43.71
-${ts} 2 1270.0.01/1234 reply ChEcKiP.DyNdNs.OrG is 91.198.22.70
-${ts} 2 1270.0.01/1234 reply ChEcKiP.DyNdNs.OrG is 216.146.43.70
-${ts} 3 1270.0.01/1234 query[A] pi.hole from 10.8.0.2
-${ts} 3 1270.0.01/1234 /etc/pihole/local.list pi.hole is 192.168.2.10
-${ts} 4 1270.0.01/1234 query[A] example.com from 10.8.0.2
-${ts} 4 1270.0.01/1234 /etc/pihole/local.list example.com is 192.168.2.10
-${ts} 5 1270.0.01/1234 query[A] play.google.com from 192.168.2.208
-${ts} 5 1270.0.01/1234 forwarded play.google.com to 2001:1608:10:25::9249:d69b
-${ts} 5 1270.0.01/1234 forwarded play.google.com to 2001:1608:10:25::1c04:b12f
-${ts} 5 1270.0.01/1234 forwarded play.google.com to 2620:0:ccd::2
-${ts} 5 1270.0.01/1234 forwarded play.google.com to 2620:0:ccc::2
-${ts} 5 1270.0.01/1234 reply play.google.com is <CNAME>
-${ts} 5 1270.0.01/1234 reply play.l.google.com is 216.58.208.110
-${ts} 5 1270.0.01/1234 reply play.l.google.com is 216.58.208.110
-${ts} 5 1270.0.01/1234 reply play.l.google.com is 216.58.208.110
-${ts} 5 1270.0.01/1234 reply play.google.com is <CNAME>
-${ts} 6 1270.0.01/1234 query[AAAA] play.google.com from 192.168.2.208
-${ts} 6 1270.0.01/1234 forwarded play.google.com to 2620:0:ccd::2
-${ts} 6 1270.0.01/1234 reply play.l.google.com is 2a00:1450:4017:802::200e
-${ts} 7 1270.0.01/1234 query[A] blacklisted.com from 192.168.2.208
-${ts} 7 1270.0.01/1234 /etc/pihole/black.list blacklisted.com is 1.2.3.4
-${ts} 8 1270.0.01/1234 query[A] addomain.com from 192.168.2.208
-${ts} 8 1270.0.01/1234 /etc/pihole/gravity.list addomain.com is 1.2.3.4
-EOT
-touch "pihole-FTL.log"
+# Install necessary additional components for testing
+apt-get -qq install dnsutils -y > /dev/null
 
-cat <<EOT >> pihole-FTL.conf
-DBFILE=pihole-FTL.db
-LOGFILE=pihole-FTL.log
-SOCKETFILE=pihole-FTL.sock
-EOT
+# Create necessary directories
+mkdir -p /etc/pihole /var/run/pihole /var/log
+
+# Prepare gravity database
+sqlite3 /etc/pihole/gravity.db < test/gravity.db.schema
+
+# Prepare setupVars.conf
+echo "BLOCKING_ENABLED=true" > /etc/pihole/setupVars.conf
+
+# Prepare pihole-FTL.conf
+echo "" > /etc/pihole/pihole-FTL.conf
 
 # Start FTL
-./pihole-FTL travis-ci
+if ! ./pihole-FTL; then
+  echo "pihole-FTL failed to start"
+  exit 1
+fi
 
 # Prepare BATS
 mkdir -p test/libs
-git submodule add https://github.com/sstephenson/bats test/libs/bats
-git submodule add https://github.com/ztombol/bats-support test/libs/bats-support
-# git submodule add https://github.com/ztombol/bats-assert test/libs/bats-assert
+git clone --depth=1 https://github.com/sstephenson/bats test/libs/bats > /dev/null
```

Comment:
This repo is the current home of bats, so it should be used:
https://github.com/bats-core/bats-core#background